packages/aws-cdk-lib/aws-stepfunctions/lib/states/distributed-map/item-reader.ts

```diff
@@ -127,15 +160,29 @@ export class S3ObjectsItemReader implements IItemReader {
    * Compile policy statements to provide relevent permissions to the state machine
    */
   public providePolicyStatements(): iam.PolicyStatement[] {
+    if (!this._bucket) return [];
```

Comment:
I think we still need to give a permission to list bucket, but the difference here is not to limit it to specific bucket, but make it for any bucket.

Reply:
Right - will do that in next revision - thanks!